bullet_train-super_scaffolding/lib/scaffolding/script.rb

```diff
@@ -183,7 +197,19 @@ def check_required_options_for_attributes(scaffolding_type, attributes, child, p
       puts "Adding new fields to #{child} with '#{generation_command}'".green
     end
     puts ""
-    `#{generation_command}` unless @options["skip-migration-generation"]
+
+    unless @options["skip-migration-generation"]
+      untracked_files = has_untracked_files?(git_status) ? get_untracked_files(git_status) : []
+      generation_thread = Thread.new { `#{generation_command}` }
+      generation_thread.join # Wait for the process to finish.
```

Comment:
I can't remember if this is where `system generation_command` or `spawn generation_command` is meant to be used. But this seems good.